src/components/Auth/SignupForm.tsx

```diff
@@ -1,33 +1,48 @@
 import PersonalInfoInput from 'components/common/PersonalInfoInput';
-import React, { useState } from 'react';
+import React, { useEffect } from 'react';
 import styled from 'styled-components';
+import { theme } from 'styles/theme';
+import { UserSignupRequest } from 'types/user';
 
-function SignupForm() {
-  const [signupInfo, setSignupInfo] = useState({
-    name: '',
-    nickname: '',
-    email: '',
-    emailConfirm: '',
-    password: '',
-    passwordConfirm: '',
-  });
+interface SignupFormProps {
+  signupInfo: UserSignupRequest;
+  handleOnChange: (type: keyof UserSignupRequest, value: string) => void;
+  handleComplete: (type: keyof UserSignupRequest, value: boolean) => void;
+}
+
+function SignupForm(props: SignupFormProps) {
+  const { signupInfo, handleOnChange, handleComplete } = props;
+  const checkPassword = () => signupInfo.password.value !== signupInfo.passwordConfirm.value;
+
+  const InfoList = Object.keys(signupInfo);
 
-  const handleOnChange = (type: string, value: string) => {
-    setSignupInfo({ ...signupInfo, [type]: value });
-  };
+  useEffect(() => {
+    signupInfo.passwordConfirm.value &&
+      !checkPassword() &&
+      handleComplete('passwordConfirm', !checkPassword());
+  }, [checkPassword()]);
```

Comment:
```suggestion
  useEffect(() => {
    if (signupInfo.passwordConfirm.value && isPasswordEqual())
        handleComplete('passwordConfirm', true);
  }, [isPasswordEqual()]);
```

이 경우에는 && 연산자보다는 if 문을 활용하는게 가독성이 더 좋다고 생각합니다!